File: Csharp/CodeWars/src/Resolutions/HelpTheBookseller.cs
namespace CodeWars.Resolutions;

public class HelpTheBookseller
{
    public static string StockSummary(string[] lstOfArt, string[] lstOf1stLetter)
    {
        if (lstOf1stLetter.Length == 0 || lstOfArt.Length == 0)
            return string.Empty;

        List<string> result = [];

        foreach (var letter in lstOf1stLetter)
        {
            string[] startedWith = [.. lstOfArt.Where(x => x.StartsWith(letter))];
            int sum = startedWith.Sum(x => Convert.ToInt32(x.Split(' ')[1]));

            result.Add($"({letter} : {sum})");
        }

        return string.Join(" - ", result);
    }
}

File: TypeScript/Codewars/src/resolutions/Help-the-bookseller.ts
export const stockList = (listOfArt: string[], listOfCat: string[]): string => {
	if (listOfArt.length === 0 || listOfCat.length === 0) {
		return "";
	}

	const output: string[] = [];

	for (const cat of listOfCat) {
		let sum = 0;

		for (const art of listOfArt) {
			const [stockListWord, value] = art.split(" ");

			if (stockListWord.startsWith(cat)) sum += Number(value);
		}

		output.push(`(${cat} : ${sum})`);
	}

	return output.join(" - ");
};
